tests/test_runner/tests.py

```diff
@@ -734,3 +736,20 @@ def test_run_checks_passes_and_teardown_raises(self):
                     runner.run_tests(['test_runner_apps.sample.tests_sample.TestDjangoTestCase'])
             self.assertTrue(teardown_databases.called)
             self.assertFalse(teardown_test_environment.called)
+
+
+class DiscoverRunnerDeprecationTests(SimpleTestCase):
+    def test_extra_tests_deprecated(self):
+        with mock.patch('django.test.runner.DiscoverRunner.setup_test_environment'), \
+                mock.patch('django.test.runner.DiscoverRunner.setup_databases'), \
+                mock.patch('django.test.runner.DiscoverRunner.run_checks'), \
+                mock.patch('django.test.runner.DiscoverRunner.teardown_test_environment'):
```

Comment:
Also, if you're already creating a custom class, can you just override the methods instead of mocking / patching?

Reply:
Goodness, that will be much less ugly!